### PR TITLE
Mac cleanup

### PR DESCRIPTION
**Pull request by jtanx**
* Clean up a bunch of old/irrelevant/out of date mac build files.
* Flattened the makefile structure under the osx folder (the recursive structure is inefficient).
* 'Fixes' the shortcut hotkey display in the least amount of effort possible. Can't say I'm too impressed with how it's been implemented...

### Type of change

- **Non-breaking change**

Supersedes #3741

### CODE REVIEW

File: fontforgeexe/startui.c [hunk 1]
```diff
@@ -1051,34 +1049,25 @@ int fontforge_main( int argc, char **argv ) {
...
-	if ( !did_keybindings ) {
-	    /* Nope. we can't. Fall back to the normal stuff */
-#endif
-	    GMenuSetShortcutDomain("FontForge-MenuShortCuts");
-	    bind_textdomain_codeset("FontForge-MenuShortCuts","UTF-8");
-	    bindtextdomain("FontForge-MenuShortCuts", getLocaleDir());
-#if defined(__Mac)
 	}
     }
 #endif
```
Comment (skef) [inline fontforgeexe/startui.c]: Given the nesting a `// defined(__Mac)` would be appreciated here.

Comment (jtanx) [reply fontforgeexe/startui.c]: Done

**Comment by jtanx** [comment]
With regards to the osx makefile, yes; everything that the previous recursive makefile structure did under the `osx` folder is now done by `osx/Makefile.am`

**Comment by frank-trampe** [comment]
I like these added/removed numbers.

**Pull request merged**